src/action/network.rs

```diff
@@ -0,0 +1,296 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! A handler and associated types for the network connection action.
+//!
+//! The network connection action lists all the opened connections on the
+//! client and the information about them (e.g. type of the connection, source
+//! and target IP addresses, the process that owns the connection)
+
+use std::fmt::{Display, Formatter};
+use std::net::IpAddr;
+
+use log::error;
+use sysinfo::{System, SystemExt, Process, ProcessExt};
+use netstat2::{
+    self,
+    ProtocolSocketInfo::{self, Tcp, Udp},
```

Comment:
`Tcp` and `Udp` are used only for pattern matching in one place. In such cases I prefer to do a local import to avoid cluttering the global namespace.